lib/fluent/plugin/in_amqp.rb

```diff
@@ -100,20 +103,25 @@ def multi_workers_ready?
     end
 
     private
-    def parse_payload(msg)
+    def parse_payload(msg, meta)
+      parsed = nil
       if @parser
-        parsed = nil
         @parser.parse msg do |_, payload|
           if payload.nil?
             log.warn "failed to parse #{msg}"
-            parsed = { "message" => msg }
+            parsed = { 'message' => msg }
           else
             parsed = payload
           end
         end
-        parsed
       else
-        { "message" => msg }
+        parsed = { 'message' => msg }
+      end
+      if @include_headers
+        log.debug 'Adding headers'
+        { 'headers' => meta[:headers] }.merge(parsed)
```

Comment:
What happens if `headers` already exists in the incoming message body and isn't a hash? Do we perhaps need to allow the target field to be configurable, or simply explicitally state that any existing field value under `headers` will be merged (if hash) or replaced entirely (if anything else)?

Reply:
@maheshkreddy - Can you let me know what you think would be best?

I'd probably go with simplicity and stick with the merge but update the docs along with it?

Reply:
@warmfusion I agree with your suggestion of allowing to have configurable field to set header field name. But, as i am on crunch with existing projects, I will go ahead and update the docs for now.

Reply:
@warmfusion Updated the docs